src/libraries/Microsoft.PowerFx.Core/Public/Config/ReadOnlySymbolTable.cs

```diff
@@ -57,7 +57,8 @@ public string DebugName
         }
 
         // Should remove. 
-        // https://github.com/microsoft/Power-Fx/issues/828
+        // https://github.com/microsoft/Power-Fx/issues/828
+        [Obsolete("Use Composition instead of Parent Pointer")]
```

Comment:
>Use Composition instead of Parent Pointer [](http://example.com/codeflow?start=19&length=41)

How does this composition work? Does symbol lookup work hierarchically?

Reply:
Yes, it's a hierarchy. 
All the SymbolTable derived classes implement INameResolver. 

The key is ComposedReadOnlySymbolTable : 

https://github.com/microsoft/Power-Fx/blob/59c8a9d6fd34c7f9476eac96c36e34d469969267/src/libraries/Microsoft.PowerFx.Core/Public/Config/ComposedReadOnlySymbolTable.cs#L120

I updated the comment in the related issue (#828) with more explanation.